build/common.sh

```diff
@@ -796,7 +798,7 @@ function kube::release::create_docker_images_for_server() {
         ln ${binary_dir}/${binary_name} ${docker_build_path}/${binary_name}
         printf " FROM ${base_image} \n ADD ${binary_name} /usr/local/bin/${binary_name}\n" > ${docker_file_path}
 
-        local docker_image_tag=gcr.io/google_containers/$binary_name:$md5_sum
+        local docker_image_tag=gcr.io/google_containers/${binary_name}-${arch}:${md5_sum}
```

Comment:
Probably, but docker is far away from that right now.
@fgrzadkowski already fixed it and the naming convention is `binary-arch:version`
It's just that he forgot to fix the development image. Some lines down, he has fixed it already for the release images